model/adapter.py

```diff
@@ -51,14 +63,23 @@ def get_existing_metabolite(mnx_id, model, compartment):
     """
     if not mnx_id:
         return
+    response = await query_identifiers([mnx_id], 'mnx', model.notes['namespace'])
+    # TODO: so messy, adjust when fixing #30
```

Comment:
what is #30?
would it be ok to check for mnx identifier first and save a call if it's found?
```python
if model.metabolites.has_id(mnx_id + compartment):
    logger.info('found {mnx_id} in {model_metabolite.id}')
    return model.metabolites.get_by_id(mnx_id + compartment)
response = await query_identifiers([mnx_id], 'mnx', model.notes['namespace'])
model_metabolite = None
try:
    logger.info('trying any of {} in {}'.format(', '.join(response[mnx_id]), compartment))
    model_metabolite = get_only_element(model.metabolites.query(
        lambda met: re.sub('_[cepm]$', '', met.id) in response[mnx_id] and met.compartment == compartment[1:]
    ))
    logger.info(f'mapped {mnx_id} to {model_metabolite.id}')
except (KeyError, IndexError):
    logger.info(f'failed to resolve {mnx_id} in {compartment} for {model.id}')
return model_metabolite
```